database/src/main/java/com/firebase/ui/database/FirebaseIndexListAdapter.java

```diff
@@ -54,6 +56,6 @@ public FirebaseIndexListAdapter(Activity activity,
                                     @LayoutRes int modelLayout,
                                     Query keyRef,
                                     Query dataRef) {
-        super(activity, modelClass, modelLayout, new FirebaseIndexArray(keyRef, dataRef));
+        super(activity, modelClass, modelLayout, new FirebaseIndexArray(keyRef, dataRef.getRef()));
```

Comment:
Probably missing something, but why accept a `Query` if we are going to call `getRef()` on it immediately? Also it seems strange to introduce a behavior change at the same time as a deprecation.